sycl/include/CL/sycl/exception.hpp

```diff
@@ -73,36 +77,57 @@ using async_handler = function_class<void(cl::sycl::exception_list)>;
 
 class runtime_error : public exception {
 public:
-  runtime_error(const char *str, cl_int err = CL_SUCCESS)
-      : exception(str, err) {}
-  runtime_error(const std::string &str, cl_int err = CL_SUCCESS)
-      : runtime_error(str.c_str(), err) {}
+  runtime_error() = default;
+
+  runtime_error(const char *Msg, cl_int Err = CL_SUCCESS)
+      : runtime_error(string_class(Msg), Err) {}
+
+  runtime_error(const string_class &Msg, cl_int Err = CL_SUCCESS)
```

Comment:
Probably you need to use pi counterparts here.

Reply:
Yes, in the future I will make it.